packages/suite/src/middlewares/wallet/fiatRatesMiddleware.ts

```diff
@@ -51,23 +51,19 @@ const fiatRatesMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => (next: D
         case ACCOUNT.CREATE: {
             // fetch current rates for account's tokens
             const account = action.payload;
-            if (account.tokens) {
-                account.tokens.forEach((t, i) => {
-                    if (t.symbol) {
-                        const s = t.symbol;
-                        // wait 500ms before firing next fetch
-                        setTimeout(() => {
-                            api.dispatch(
-                                fiatRatesActions.updateCurrentRates({
-                                    symbol: s,
-                                    mainNetworkSymbol: account.symbol,
-                                    tokenAddress: t.address,
-                                }),
-                            );
-                        }, i * 500);
-                    }
-                });
-            }
+            account.tokens?.forEach(t => {
+                if (!t.symbol) {
+                    return;
+                }
+                api.dispatch(
+                    fiatRatesActions.updateCurrentRates({
```

Comment:
We start to update data periodically. However, we do not stop this when device is disconnected or coin having tokens is disabled.